### PR TITLE
refactor: replace PowerBI config file content by data from Workspace

### PR DESCRIPTION
**Pull request by csm-thu**
BREAKING CHANGES: PowerBI.js configuration file format has changed. Scenario view report and Dashboards view reports must now be configured in the Workspace object declared to Cosmotech API, in:
 - [workspace].webapp.options.charts.dashboardsView
 - [workspace].webapp.options.charts.scenarioView

Reports configuration data can no longer use
`PowerBIReportEmbedMultipleFilter` and `PowerBIReportEmbedSimpleFilter` objects for filters, each filter must now be described by a simple object.

Example:
`new PowerBIReportEmbedMultipleFilter('Bar', 'Bar', ['MyBar', 'MyBar2'])`

must be replaced by:

`{table: 'Bar', column: 'Bar', values: ['MyBar', 'MyBar2']}`

For `PowerBIReportEmbedSimpleFilter`, a single value can be provided for the "values" key:
`new PowerBIReportEmbedSimpleFilter('StockProbe', 'SimulationRun', 'csmSimulationRun')`

must be replaced by:
`{table: 'StockProbe', column: 'SimulationRun', values: 'csmSimulationRun'}`

### CODE REVIEW

**Review by ardvi — APPROVED** [review]
LGTM

**Review by esasova — APPROVED** [review]
LGTM :+1:

File: api/GetEmbedInfo/embedConfigService.js [hunk 1]
```diff
@@ -25,15 +25,16 @@ async function getEmbedInfo() {
...
       accesses: null,
       error: {
         status: err.status,
         statusText: err.statusText,
-        powerBIErrorInfo: err.headers.get('x-powerbi-error-info'),
+        powerBIErrorInfo: err.headers?.get('x-powerbi-error-info'),
         description:
           `Error while retrieving report embed details\r\n${err.statusText}\r\nRequestId: \n` +
-          `${err.headers.get('requestid')}`,
+          `${err.headers?.get('requestid')}`,
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply api/GetEmbedInfo/embedConfigService.js:37]: I think `err.headers?.get` can return `undefined` in two cases:
- if the error caught doesn't have a `headers` property (i.e. if we catch a non-network exception)
- if the error caught has a `headers` property but doesn't have the field "requestid" in their headers

In these cases, the backtick + curly braces string-formatting will return a string containing "undefined", that will be concatenated with the first line. So I guess that in the worst case, we'll see `RequestId: undefined` in the error description

[the remark this answers is not in the document]
Comment (esasova) [reply api/GetEmbedInfo/embedConfigService.js:37]: Thanks !

**Comment by csm-thu** [comment]
Branch manually merged in **feat/multi_workspace**, closing this PR

**Comment by csm-thu** [comment]
Closing PR again, don't mind me

**Review by nborde-CSM — APPROVED** [review]
Some remarks but LGTM :+1:

File: src/managers/DashboardsManager/DashboardsManager.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { useDashboardsManager } from './DashboardsManagerHook';
+
+const DashboardsManager = () => {
+  useDashboardsManager();
+  return null;
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply src/managers/DashboardsManager/DashboardsManager.js:8]: This manager is a React component with nothing to render, so we return `null`. We use a React component to simplify access to redux and usage of custom hooks :slightly_smiling_face:

File: src/config/overrides/Workspaces.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+const defaultScenarioViewReport = {
+  title: { en: 'Scenario dashboard', fr: 'Rapport du scenario' },
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply src/config/overrides/Workspaces.js:5]: Reports titles in the scenario view are never displayed anyway :stuck_out_tongue: (but I'll fix this)

**Pull request closed without merging**